components/sidebar_right/sidebar_right.tsx

```diff
@@ -166,9 +168,8 @@ export default class SidebarRight extends React.PureComponent<Props, State> {
             this.props.actions.setRhsExpanded(false);
         }
 
-        // close when changing teams, close when changing products
+        // close when changing products
         if (
-            this.props.teamId !== prevProps.teamId ||
```

Comment:
This functionality is already present in https://github.com/mattermost/mattermost-webapp/blob/6caf3dcbafc309f0fc24bdb5b708d4452caf8a0a/components/needs_team/needs_team.tsx#L226-L228 . And the identification of team and closing RHS is caught way top in parent component.